library/init/category/cont.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2018 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sebastian Ullrich
+
+The continuation monad transformer.
+-/
+
+prelude
+import init.category.alternative init.category.combinators init.category.lift
+universes u v w
+
+/-- An implementation of [ContT](https://hackage.haskell.org/package/mtl-2.2.2/docs/Control-Monad-Cont.html#t:ContT) -/
+structure cont_t (ρ : Type u) (m : Type u → Type v) (α : Type u) : Type (max u v) :=
+(run : (α → m ρ) → m ρ)
+
+attribute [pp_using_anonymous_constructor] cont_t
+
+/-- An implementation of [MonadCont done right](https://wiki.haskell.org/MonadCont_done_right) -/
+class monad_cont (m : Type u → Type v) :=
```

Comment:
Why do we need this type class?
I am assuming we have `monad_except` because we want to have a general `throw` and `catch` signature.
AFAICT, we will not have multiple implementations of `call_cc`. 
Note that we don't have a general `put`, `get` and `read` operations.